.github/workflows/az-deploy.yml

```diff
@@ -0,0 +1,37 @@
+name: Deploy Azure Resources
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
+
+      - name: Setup Azure CLI
+        run: |
+          curl -sL https://aka.ms/InstallAzureCLIDeb | sudo bash
+          az --version  # Verify installation
+
+      - name: Login to Azure
+        run: |
+          az login --service-principal -u ${{ secrets.AUTO_AZURE_CLIENT_ID }} -p ${{ secrets.AUTO_AZURE_CLIENT_SECRET }} --tenant ${{ secrets.AUTO_AZURE_TENANT_ID }}
+          az account set --subscription ${{ secrets.AUTO_AZURE_SUBSCRIPTION_ID }}
+
+      - name: Install Bicep CLI
+        run: az bicep install
+
+      - name: Deploy Bicep Template
+        id: deploy
+        run: |
+          set -e
+          az deployment sub create \
+            --name automation-CWYD \
+            --location eastus \
+            --template-file infra/main.bicep \
```

Comment:
If this is to test the one click deploy button then I think this should use the `infra/main.json` file instead of the bicep file.

Reply:
There are basically two ways:
1. we can use directly bicep files to deploy the resource
2.  or else first we need to generate arm template (nothing but main.json file) from bicep file. after it we can deploy the resources.
So, we are using step 1, instead of creating arm templates, we directly using bicep file

Is there any specific reason that you are asking to use main.json instead of bicep module? Please let us know.